[expo] Build broken with SDK 46

- [x] I have updated Purchases SDK to the latest version (`"react-native-purchases": "^4.6.1",`)
- [x] I have read the [Contribution Guidelines](https://github.com/RevenueCat/react-native-purchases/blob/main/CONTRIBUTING.md)
- [x] I have searched the [Community](https://community.revenuecat.com)
- [x] I have read [docs.revenuecat.com](https://docs.revenuecat.com/)
- [x] I have searched for [existing Github issues](https://github.com/RevenueCat/react-native-purchases/issues)

**Describe the bug**
I upgraded from Expo SDK 44 to Expo SDK 46, and the EAS build gets broken.

```
eas build -p android --auto-submit --clear-cache
```

Result:
```
(...)
[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-purchases/android/src/main/java/com/revenuecat/purchases/react/RNPurchasesConverters.kt: (82, 42): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.6.0, expected version is 1.4.0.
[stderr] 
The class is loaded from /home/expo/.gradle/caches/transforms-3/f9a6c4180d9d810f143b823eeeb39ba3/transformed/jetified-kotlin-stdlib-1.6.10.jar!/kotlin/Unit.class
[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-purchases/android/src/main/java/com/revenuecat/purchases/react/RNPurchasesConverters.kt: (83, 45): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.6.0, expected version is 1.4.0.
[stderr] 
The class is loaded from /home/expo/.gradle/caches/transforms-3/f9a6c4180d9d810f143b823eeeb39ba3/transformed/jetified-kotlin-stdlib-1.6.10.jar!/kotlin/Unit.class
[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-purchases/android/src/main/java/com/revenuecat/purchases/react/RNPurchasesConverters.kt: (84, 44): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.6.0, expected version is 1.4.0.
[stderr] 
The class is loaded from /home/expo/.gradle/caches/transforms-3/f9a6c4180d9d810f143b823eeeb39ba3/transformed/jetified-kotlin-stdlib-1.6.10.jar!/kotlin/Unit.class
[stderr] 
e: /home/expo/workingdir/build/node_modules/react-native-purchases/android/src/main/java/com/revenuecat/purchases/react/RNPurchasesConverters.kt: (85, 43): Class 'kotlin.Unit' was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.6.0, expected version is 1.4.0.
[stderr] 
The class is loaded from /home/expo/.gradle/caches/transforms-3/f9a6c4180d9d810f143b823eeeb39ba3/transformed/jetified-kotlin-stdlib-1.6.10.jar!/kotlin/Unit.class
[stderr] 
FAILURE: Build failed with an exception.
[stderr] 
* What went wrong:
[stderr] 
Execution failed for task ':react-native-purchases:compileReleaseKotlin'.
```

1. Environment
   1. Platform:  Running on `eas` servers
   2. SDK version: Expo 46
   3. OS version: Running on `eas` servers
   4. Xcode/Android Studio version: Running on `eas` servers
   5. React Native version: `0.69.3`
   6. SDK installation (CocoaPods + version or manual): Running on `eas` servers
   7. How widespread is the issue. Percentage of devices affected.: Build
2. [Debug logs](https://docs.revenuecat.com/docs/debugging) that reproduce the issue
3. Steps to reproduce, with a description of expected vs. actual behavior
4. **Other information** (e.g. stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, etc.)

**Additional context**
...

--- a/android/gradle.properties
+++ b/android/gradle.properties
@@ -1,5 +1,5 @@
 android.useAndroidX=true
-Purchases_kotlinVersion=1.5.31
+Purchases_kotlinVersion=1.6.21
 Purchases_compileSdkVersion=28
 Purchases_targetSdkVersion=28
 Purchases_minSdkVersion=16

--- a/android/src/main/java/com/revenuecat/purchases/react/RNPurchasesConverters.kt
+++ b/android/src/main/java/com/revenuecat/purchases/react/RNPurchasesConverters.kt
@@ -53,6 +53,7 @@ internal object RNPurchasesConverters {
         return array
     }
 
+    @Suppress("UNCHECKED_CAST")
     private fun convertArrayToWritableArray(array: Array<Any?>): WritableArray {
         val writableArray: WritableArray = WritableNativeArray()
         for (item in array) {
@@ -71,6 +72,7 @@ internal object RNPurchasesConverters {
     }
 
     @JvmStatic
+    @Suppress("UNCHECKED_CAST")
     fun convertMapToWriteableMap(map: Map<String, *>): WritableMap {
         val writableMap: WritableMap = WritableNativeMap()
         for ((key, value) in map) {

